feat(ISV-5885): update release-time SBOM documentation (#469)
konflux-ci/docs: modules/metadata/pages/sboms.adoc

You can use an SBOM to better understand the composition of your software. In particular, you can use various tools to scan your SBOM and determine if any of your dependencies have known vulnerabilities. Furthermore, to comply with cybersecurity regulations, your organization might need to provide its customers with the SBOM for your application's components. For example, in the United States, the link:https://www.whitehouse.gov/briefing-room/presidential-actions/2021/05/12/executive-order-on-improving-the-nations-cybersecurity/[Executive Order on Improving the Nation's Cybersecurity] requires vendors to provide SBOMs, and in Europe, the link:https://www.cisa.gov/sites/default/files/2023-09/EU%20Commission%20SBOM%20Work_508c.pdf[Cyber Resilience Act] does the same.

== Types of SBOMs produced by {ProductName}

> {ProductName} produces *build-time* and *release-time* SBOMs. *Build-time* SBOMs are created directly during the build phase of the component. *Release-time* SBOMs are augmented with additional information which is known only during the release phase of the component. This includes information about the repository to which the component will be released.

Reviewer: ```suggestion
{ProductName} produces *build-time* and *release-time* SBOMs. *Build-time* SBOMs are created directly during the build phase of the component. *Release-time* SBOMs are augmented with additional information which is known only during the release phase of the component. This includes information about the repository the component was released to.
```
